### DISCUSSION TITLE
UInt keys not working for Maps on Algorand ?

### DISCUSSION THREAD

**Discussion by asolpshinning** [reach-sh/reach-lang#1198] [Q&A] [answered]
Is this true  or do I have an outdated compiler? The documentation did not warn about this at this link https://docs.reach.sh/rsh/consensus/#p_83
I have made plans on using a UInt key for maps. It will be disappointing if it is not possible on Algorand. Got this error:

`Error: This Reach application is not supported on Algorand for the following reasons:`
 `Cannot use 'UInt' as Map key. Only 'Address' keys are allowed.` 

If true, please kindly tell me a work around using `UInt` as map key on Algorand.

**Comment by jeapostrophe** [comment, marked as answer]
That's true. Algorand doesn't support things like that.

**Reply by jeapostrophe** [reply]
They expect to add it this summer

**Reply by asolpshinning** [reply]
Thanks... Waiting for it. Really important
